### PR TITLE
Add sample: Front Door Standard/Premium (Preview) with Application Gateway origin

### PR DESCRIPTION
**Pull request by johndowns**
# PR Checklist

Check these items before submitting a PR... 

[Contribution Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md)

[Best Practice Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md)

- [X] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

## Changelog

* Add sample - Front Door Standard/Premium (Preview) with Application Gateway origin

### CODE REVIEW

File: 201-front-door-standard-premium-application-gateway-public/prereqs/prereq.azuredeploy.bicep [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+  name: guid(resourceGroup().id, managedIdentityName, storageAccountContributorRoleDefinitionId)
+  properties: {
+    roleDefinitionId: storageAccountContributorRoleDefinitionId
+    principalId: managedIdentity.properties.principalId
+  }
+}
+
+resource roleAssignmentStorageBlobDataContributor 'Microsoft.Authorization/roleAssignments@2020-04-01-preview' = {
+  scope: storageAccount
+  name: guid(resourceGroup().id, managedIdentityName, storageAccountStorageBlobDataContributorRoleDefinitionId)
```
Comment (bmoore-msft) [inline 201-front-door-standard-premium-application-gateway-public/prereqs/prereq.azuredeploy.bicep:56]: managedIdentityName should be the resourceId for the identity

File: 201-front-door-standard-premium-application-gateway-public/modules/application-gateway.bicep [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+    dnsSettings: {
+      domainNameLabel: publicIPAddressDomainNameLabel
+    }
+  }
+}
+
+resource applicationGateway 'Microsoft.Network/applicationGateways@2019-09-01' = {
+  name: applicationGatewayName
+  location: location
+  zones: []
```
[the remark this answers is not in the document]
Comment (johndowns) [reply 201-front-door-standard-premium-application-gateway-public/modules/application-gateway.bicep:57]: Fixed.

File: 201-front-door-standard-premium-application-gateway-public/prereqs/prereq.azuredeploy.bicep [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+}
+
+resource managedIdentity 'Microsoft.ManagedIdentity/userAssignedIdentities@2018-11-30' = {
+  name: managedIdentityName
+  location: location
+}
+
+resource roleAssignmentContributor 'Microsoft.Authorization/roleAssignments@2020-04-01-preview' = {
+  scope: storageAccount
+  name: guid(resourceGroup().id, managedIdentityName, storageAccountContributorRoleDefinitionId)
```
[the remark this answers is not in the document]
Comment (johndowns) [reply 201-front-door-standard-premium-application-gateway-public/prereqs/prereq.azuredeploy.bicep]: Fixed.

File: 201-front-door-standard-premium-application-gateway-public/prereqs/prereq.azuredeploy.bicep (hunk 1 above)
Comment (johndowns) [reply 201-front-door-standard-premium-application-gateway-public/prereqs/prereq.azuredeploy.bicep]: Fixed.

**Comment by bmoore-msft** [comment]
@johndowns - check this PR for updates that may be needed to documentation that references this sample.  [this is an automated message]

**Pull request merged**